src/Compilers/CSharp/Test/Emit2/Semantics/LockTests.cs

```diff
@@ -0,0 +1,1102 @@
+        //             public void Dispose() { }
+        //         }
+        //     }
+        // }
+        var ilSource = """
+            .class public auto ansi sealed beforefieldinit System.Threading.Lock extends System.Object
+            {
+                .method public hidebysig specialname rtspecialname instance void .ctor() cil managed
+                {
+                    .maxstack 8
+                    ret
+                }
+
+                .method public hidebysig instance class System.Threading.Lock/Scope EnterLockScope () cil managed
+                {
+                    .maxstack 8
+                    ldnull
+                    throw
+                }
+
+                .class nested public sequential ansi sealed beforefieldinit Scope extends System.ValueType
+                {
+                    .custom instance void System.Runtime.CompilerServices.IsByRefLikeAttribute::.ctor() = (
+                        01 00 00 00
+                    )
+
+                    .pack 0
+                    .size 1
+
+                    .method public hidebysig instance void Dispose () cil managed
+                    {
+                        .custom instance void System.Runtime.CompilerServices.CompilerFeatureRequiredAttribute::.ctor(string) = (
+                            01 00 04 54 65 73 74 00 00
+                        )
+                        .maxstack 8
+                        ret
+                    }
+                }
+            }
+
+            .class public auto ansi sealed beforefieldinit System.Runtime.CompilerServices.CompilerFeatureRequiredAttribute extends System.Object
+            {
+                .method public hidebysig specialname rtspecialname instance void .ctor(string s) cil managed
+                {
+                    .maxstack 8
+                    ret
+                }
+            }
+
+            .class public auto ansi sealed beforefieldinit System.Runtime.CompilerServices.IsByRefLikeAttribute extends System.Object
+            {
+                .method public hidebysig specialname rtspecialname instance void .ctor() cil managed
+                {
+                    .maxstack 8
+                    ret
+                }
+            }
+            """;
+        var source = """
+            class C
+            {
+                void M(System.Threading.Lock l)
+                {
+                    lock (l) { }
+                }
+            }
+            """;
+        CreateCompilationWithIL(source, ilSource).VerifyEmitDiagnostics(
+            // (5,15): error CS9041: 'Lock.Scope.Dispose()' requires compiler feature 'Test', which is not supported by this version of the C# compiler.
+            //         lock (l) { }
+            Diagnostic(ErrorCode.ERR_UnsupportedCompilerFeature, "l").WithArguments("System.Threading.Lock.Scope.Dispose()", "Test").WithLocation(5, 15));
+    }
+}
```

Comment:
We should test scenario with multiple distinct `Lock` types in the picture at the same time. Given the current design choice, all of them should be handled specially.